### ISSUE TITLE
[1.16.1] Relogging doesn't change the player profile

### ISSUE THREAD

**Issue by MrIvanPlays** [PaperMC/Paper#3841]
### What behaviour is expected:
I have a method which calls the player profile api on PlayerLoginEvent so that a skin set is triggered and it should do that.

### What behaviour is observed:
It does what it is intended to do, but only for the first login. After you relog, it doesn't set it again.

### Steps/models to reproduce:
This getting called in PlayerLoginEvent:
```java
    PlayerProfile profile = player.getPlayerProfile();
    profile.setProperty(new ProfileProperty("textures", skin.getTexture(), skin.getSignature()));
    player.setPlayerProfile(profile);
```

### Plugin list:
irrelevant

### Paper build number:
This server is running Paper version git-Paper-78 (MC: 1.16.1) (Implementing API version 1.16.1-R0.1-SNAPSHOT)

### Anything else:
It is not mojang api rate limit cuz I print out the skin before I set it, and everything seems to be fine with the texture and the signature.
Yes I am aware of AsyncPlayerPreLoginEvent having the profile exposed too, however I have to do database updates before calling this code (its not my database updates not being finished so it doesn't call the code, it IS being called I checked that too).

**Comment by aikar** [comment]
are you changing the skin each time? if so, its probably a client cache.
if the client shares same cache skulls use, then it is absolutely that, but should be fixed in 1.16.2

**Comment by MrIvanPlays** [comment]
The weird thing is that this worked in git-Paper-10. The skin stays the same right now. 
Honestly idk what's wrong, I literally have prints everywhere, and I see it calling the profile api. 

EDIT: Not too sure but this may be the commit from which started that problem: https://github.com/PaperMC/Paper/commit/49491f32d9774086c4b89f812812e54cb12b2140

I'm going to revert it and see what happens

**Comment by MrIvanPlays** [comment]
Ok now I don't know whats wrong.. is it paper or is it my plugin screwd up... Printing shows that everything is fine, when its actually not...

**Comment by MrIvanPlays** [comment]
Ok so it looks like it works properly with your own skin, but it doesn't work with a custom skin of the account you're currently on (online-mode=false). 
@aikar perhaps you're right its the client cache issue. I'll still leave this open tho

**Comment by aikar** [comment]
as said if its client cache, its fixed in 1.16.2 and otherwise not fixable by us.

**Comment by MrIvanPlays** [comment]
Just to confirm for the future, this is definitely a client issue, cuz on 1.15.2 everything seems normal with the exact same code I have.
